src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -576,17 +576,12 @@ public Type ReadType()
         private Type ReadTypeAfterTag()
             => _binderSnapshot.GetTypeFromId(this.ReadInt32());
 
-        private Func<ObjectReader, object> ReadTypeReader()
-        {
-            _reader.ReadByte();
-            return _binderSnapshot.GetTypeReaderFromId(this.ReadInt32());
-        }
-
         private object ReadObject()
         {
             var id = _objectReferenceMap.GetNextReferenceId();
 
-            var typeReader = this.ReadTypeReader();
+            _reader.ReadByte();
```

Comment:
❕ Please add a comment so it's clear what this byte is.

❔ This appears to be the `EncodingKind.Object` value written in `WriteObjectWorker`. Is it used elsewhere? If not we might want to remove it.